test/utils.test.ts

```diff
@@ -1,50 +1,155 @@
             expect(compareTextWithArray(' foo ', ['foo', 'bar'], { trim: true }).result).toBe(true)
         })
 
-        test('ignoreCase', () => {
+        test('should pass if wrong case and using ignoreCase', () => {
             expect(compareTextWithArray(' FOO ', ['foO', 'bar'], { trim: true, ignoreCase: true }).result).toBe(true)
         })
 
-        test('containing', () => {
+        test('should pass if string contains and using containing', () => {
             expect(compareTextWithArray('qwe_AsD_zxc', ['foo', 'zxc'], { ignoreCase: true, containing: true }).result).toBe(true)
         })
     })
+
+    describe('waitUntil', () => {
+        let condition: any
+        const positiveCondition = () => { return {result: true, message: "test" }}
+        const negativeCondition = () => { return {result: false, message: "test" }}
+
+        describe('positive condition', () => {
+            beforeEach(() => { 
+                condition = async () => { 
+                    return positiveCondition().result
+                } 
+            })
+
+            test('should fail if isNot is true', async () => {
+                const pass = await waitUntil(condition, true, {})
+                expect(pass).toBe(false)
+            })
+
+            test('should pass if isNot is false', async () => {
+                const pass = await waitUntil(condition, false, {})
+                expect(pass).toBe(true)
+            })
+            
+            test('should fail if isNot is true and wait is 0', async () => {
+                const pass = await waitUntil(condition, true, {wait: 0})
+                expect(pass).toBe(false)
+            })
+
+            test('should pass if isNot is false and wait is 0', async () => {
+                const pass = await waitUntil(condition, false, {wait: 0})
+                expect(pass).toBe(true)
+            })
+        })
+
+        describe('negative condition', () => {
+            beforeEach(() => { 
+                condition = async () => { 
+                    return negativeCondition().result
+                } 
+            })
+
+            test('should pass if isNot is true', async () => {
+                const pass = await waitUntil(condition, true, {})
+                expect(pass).toBe(true)
+            })
+
+            test('should fail if isNot is false', async () => {
+                const pass = await waitUntil(condition, false, {})
+                expect(pass).toBe(false)
+            })
+
+            test('should pass if isNot is true and wait is 0', async () => {
+                const pass = await waitUntil(condition, true, {wait: 0})
+                expect(pass).toBe(true)
+            })
+
+            test('should fail if isNot is false and wait is 0', async () => {
+                const pass = await waitUntil(condition, false, {wait: 0})
+                expect(pass).toBe(false)
+            })
+        })
+    })
```

Comment:
@christian-bromann The reason I changed the waitUntil function was because I initially wrote these test cases and some of them failed. Are these test cases the correct interpretation of isNot? Because maybe I have confused something here. 

Then having updated the waitUntil to follow this logical pattern I then noticed that some of the tests using isNot didn't seem to align with this logic and so I changed these conditions as I believed they were the wrong way around. The comments above assertions also seemed to point to this